src/KPM.jl

```diff
@@ -38,7 +38,29 @@ function matrixKPM(h::Hamiltonian{<:Lattice,L}, matrixtype = missing) where {L}
     return bloch!(m, h)
 end
 
-matrixKPM(A::UniformScaling) = A
+matrixKPM(kets::StochasticTraceKets, h::Hamiltonian) = 
+    matrixKPM(Matrix(kets, h), first(length.(orbitals(orbitalstructure(h)))))
+
+matrixKPM(ketmat::Matrix{T}, orbitals) where{T<:Number} = ketmat
+
+function matrixKPM(ketmat::Matrix{SMatrix{N,M,T,P}}, orbitals) where{N,M,T,P}
+        flatsize = size(ketmat) .* (orbitals, 1)
+        m = similarmatrix(ketmat, Matrix{T}, flatsize)
+        flatten_kets!(m, ketmat, orbitals)
+        return m
+end
+
+matrixKPM(A::UniformScaling, matrixtype = missing) = A
+
+function flatten_kets!(m, kets´´, orbitals)
+    rows, cols = size(kets´´)
+    for j in 1:cols
+        for i in 1:rows
+            m[1+(i-1)*orbitals:i*orbitals, j] = diag(kets´´[i, j])
+        end
+    end
+    return m
+end
```

Comment:
See comment above: this will likely fail if you have heterogeneous multiorbital systems

Reply:
I haven't thought of this possibility. I'll take into account this in the next update so it won't fail in those cases.